File: app/assets/javascripts/application.js
// This is a manifest file that'll be compiled into application.js, which will include all the files
// listed below.
//
// Any JavaScript/Coffee file within this directory, lib/assets/javascripts, vendor/assets/javascripts,
// or any plugin's vendor/assets/javascripts directory can be referenced here using a relative path.
//
// It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
// compiled file.
//
// Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
// about supported directives.
//
//= require jquery
//= require jquery_ujs
//= require bootstrap
//= require bootstrap-sprockets
//= require angular/angular
//= require angular-ui-router/release/angular-ui-router
//= require_self
//= require angular-sanitize
//= require angular-rails-templates
//= require ng-app/app
//= require_tree ./ng-app/templates
//= require_tree ./ng-app/modules
//= require_tree ./ng-app/filters
//= require_tree ./ng-app/directives
//= require_tree ./ng-app/models
//= require_tree ./ng-app/services
//= require_tree ./ng-app/controllers

var app = angular.module('catalyst', ['ui.router']);

File: config/application.rb
require File.expand_path('../boot', __FILE__)

require 'rails/all'

# Require the gems listed in Gemfile, including any gems
# you've limited to :test, :development, or :production.
Bundler.require(*Rails.groups)

module Workspace
  class Application < Rails::Application
    config.assets.paths << Rails.root.join("vendor", "assets", "bower_components")
    
    #Telling asset pipeline about angular directories
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "controllers")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "directives")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "filters")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "models")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "modules")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "services")
    config.assets.paths << Rails.root.join("app", "assets", "javascripts", "ng-app", "templates")
    
    config.assets.precompile += %w(app.js)
    config.assets.precompile += %w(HomeCtrl.js)
    

    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration should go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded.

    # Set Time.zone default to the specified zone and make Active Record auto-convert to this zone.
    # Run "rake -D time" for a list of tasks for finding time zone names. Default is UTC.
    # config.time_zone = 'Central Time (US & Canada)'

    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
    # config.i18n.default_locale = :de

    # Do not swallow errors in after_commit/after_rollback callbacks.
    config.active_record.raise_in_transactional_callbacks = true
  end
end
